### ISSUE TITLE
Improve noaa-predict examples

### ISSUE THREAD

**Issue by cpbridges** [sunpy/sunpy#3359]
I am following the guidance here to plot / calculate / play with solar cycle data from:
- https://docs.sunpy.org/en/stable/generated/gallery/plotting/solar_cycle_example.html?highlight=noaa-predict#plotting-the-solar-cycle-index
- https://docs.sunpy.org/en/stable/generated/gallery/plotting/solar_cycle_example.html?highlight=noaa-predict

But when I try to merge these two guidances I have an unexpected problem:

```
from sunpy.net import Fido, attrs as a
from sunpy.io.file_tools import read_file
import sunpy.timeseries as ts

result = Fido.search(a.Time('2016/1/1','2019/1/1'),a.Instrument('noaa-predict'))
downloaded = Fido.fetch(result)
print(downloaded)
data, header = read_file(downloaded[0])[0]
```
```
---------------------------------------------------------------------------
UnrecognizedFileTypeError                 Traceback (most recent call last)
<ipython-input-6-7852bf71466b> in <module>
      1 # noaa_predict = ts.TimeSeries(downloaded, source='noaapredictindices')
      2 
----> 3 data, header = read_file(downloaded[0])[0]

~/miniconda3/envs/py37/lib/python3.7/site-packages/sunpy/io/file_tools.py in read_file(filepath, filetype, **kwargs)
     86 
     87     # If filetype is not apparent from the extension, attempt to detect it
---> 88     readername = _detect_filetype(filepath)
     89     return _readers[readername].read(filepath, **kwargs)
     90 

~/miniconda3/envs/py37/lib/python3.7/site-packages/sunpy/io/file_tools.py in _detect_filetype(filepath)
    214 
    215     # Raise an error if an unsupported filetype is encountered
--> 216     raise UnrecognizedFileTypeError("The requested filetype is not currently "
    217                                     "supported by SunPy.")
    218 

UnrecognizedFileTypeError: The requested filetype is not currently supported by SunPy.
```
I did try:
```
noaa_predict = ts.TimeSeries(downloaded, source='noaapredictindices')
```
```
WARNING: ErfaWarning: ERFA function "dtf2d" yielded 12 of "dubious year (Note 6)" [astropy._erfa.core]
WARNING: ErfaWarning: ERFA function "d2dtf" yielded 12 of "dubious year (Note 5)" [astropy._erfa.core]
WARNING: ErfaWarning: ERFA function "dtf2d" yielded 1 of "dubious year (Note 6)" [astropy._erfa.core]
```
I posted in the Matrix and checked GitHub issues but could see anything similar.

I'd like to look at the sun spots to begin with, but also 10.7 cm flux and geomagnetic index also.
Which is the correct/best way forward?

Using Jupyter Notebook via conda. Sunpy 1.03. Macbook Air OSX 10.13.6.

Thanks in advance! Chris

**Comment by Cadair** [comment]
Ignoring those ERFA warnings did your:
```python
noaa_predict = ts.TimeSeries(downloaded, source='noaapredictindices')
```
example work?

The Erfa warnings are just because the dates in the predict series are far enough in the future that they are out of the range of the UTC leap second tables in Astropy.

**Comment by cpbridges** [comment]
Hi Cadair! Thanks for getting back - and good spot. I tried:
```
noaa = ts.TimeSeries(downloaded, source='noaaindices')
plt.plot(noaa.data.index, noaa.data['sunspot RI'], label='Sunspot Number')
plt.ylim(0)
plt.ylabel('Sunspot Number')
plt.xlabel('Year')
plt.legend()
plt.show()
```
But get:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-26d50cef829c> in <module>
----> 1 noaa = ts.TimeSeries(downloaded, source='noaaindices')
      2 
      3 plt.plot(noaa.data.index, noaa.data['sunspot RI'], label='Sunspot Number')
      4 # plt.plot(noaa_predict.data.index, noaa_predict.data['sunspot'],
      5 #          color='grey', label='Near-term Prediction')

~/miniconda3/envs/py37/lib/python3.7/site-packages/sunpy/timeseries/timeseries_factory.py in __call__(self, silence_errors, *args, **kwargs)
    374         for filepath in filepaths:
    375             try:
--> 376                 new_ts = self._check_registered_widgets(filepath=filepath, **kwargs)
    377                 new_timeseries.append(new_ts)
    378             except (NoMatchError, MultipleMatchError, ValidationFunctionError):

~/miniconda3/envs/py37/lib/python3.7/site-packages/sunpy/timeseries/timeseries_factory.py in _check_registered_widgets(self, **kwargs)
    497         units = kwargs.pop('units', None)
    498         if filepath:
--> 499             data, meta, units = WidgetType._parse_file(filepath)
    500 
    501         # Now return a TimeSeries from the given file.

~/miniconda3/envs/py37/lib/python3.7/site-packages/sunpy/timeseries/sources/noaa.py in _parse_file(cls, filepath)
    127                             comment='#', dtype={'yyyy': np.str, 'mm': np.str})
    128             data = data.dropna(how='any')
--> 129             timeindex = Time.strptime([x + y for x, y in zip(data['yyyy'], data['mm'])], '%Y%m')
    130             timeindex.precision = 9
    131             data['time'] = timeindex.isot.astype('datetime64')

~/miniconda3/envs/py37/lib/python3.7/site-packages/astropy/time/core.py in strptime(cls, time_string, format_string, **kwargs)
    560             err = "Expected type is string, a bytes-like object or a sequence"\
    561                   " of these. Got dtype '{}'".format(time_array.dtype.kind)
--> 562             raise TypeError(err)
    563 
    564         to_string = (str if time_array.dtype.kind == 'U' else

TypeError: Expected type is string, a bytes-like object or a sequence of these. Got dtype 'f'
```
I think this means the ts object needs some conversion... time to Google I guess!

Only Q left is where to get the 10.7 cm and geotag indices. Again, perhaps I can google these.

Chris

**Comment by Cadair** [comment]
That is a bug, I will try and look into it later.

As for the 10.7 and geomag, if you can find good sources of those data then I would be happy to add them. I also wonder if they might be supported in [heliopy](https://docs.heliopy.org/en/stable/) @dstansby ?

**Comment by Cadair** [comment]
also ping @sunpy/timeseries

**Comment by dstansby** [comment]
If it does exist in HelioPy then it will be via the OMNI dataset, which is accessible here: https://docs.heliopy.org/en/stable/data/omni.html , otherwise if not a [report in HelioPy](https://github.com/heliopython/heliopy/issues) with a feature request to add the relevant data source would be very welcome!

**Comment by hayesla** [comment]
Hey @cpbridges!

It looks like that you have downloaded the noaa-predict indices rather than the noaa indices, so when you are loading the timeseries the wrong source is labelled. 

You want to be looking at the noaa-indices as this has both the geomagnetic index and F10.7 radio flux - 

For example:
```
>>> from sunpy.net import Fido, attrs as a
>>> import sunpy.timeseries as ts

>>> result = Fido.search(a.Time('2016/1/1','2019/1/1'),a.Instrument('noaa-indices'))
>>> downloaded_file = Fido.fetch(result)

>>> noaa_indices = ts.TimeSeries(downloaded_file, source='noaaindices')
>>> print(noaa_indices.columns)
['sunspot SWO', 'sunspot RI', 'sunspot ratio', 'sunspot SWO smooth', 'sunspot RI smooth', 'radio flux', 'radio flux smooth', 'geomagnetic ap', 'geomagnetic smooth']
```
You can then plot the geomagnetic AP and F10.7 radio flux:

```
fig, ax = plt.subplots(2, sharex=True)
ax[0].plot(noaa_indices.index, noaa_indices.data['geomagnetic ap'], label='AP')
ax[0].legend()
ax[0].set_ylabel('Geomagnetic AP')
ax[1].plot(noaa_indices.index, noaa_indices.data['radio flux'], label='F10.7')
ax[1].legend()
ax[1].set_ylabel('Radio flux at 10.7cm')
ax[1].set_xlabel('Time')
plt.tight_layout()
plt.show()
```

![noaa_example](https://user-images.githubusercontent.com/4620298/65125894-8b137080-d9f6-11e9-8cc5-c01595aef738.png)

**Comment by hayesla** [comment]
also here's an example of plotting both the NOAA indices and predicted indices for the solar cycle prediction https://gist.github.com/hayesla/31749184de3e36469c1a91c0f591a562

**Comment by hayesla** [comment]
maybe we need to update the example gallery @Cadair to add in downloading the data?

**Comment by cpbridges** [comment]
@hayesla Hi Laura! I spotted the indices error but didn't explore the returned object parameters.
I did look up the API but must've missed something. 

Thanks so much to make an example! It would be good to see it public.

I'll be trying to pull the full known history to fit some curves next for my teaching :)

I'll let devs close when appropriate! Chris

**Comment by nabobalis** [comment]
We can change this issue to track updating the NOAA example(s).

**Comment by mridullpandey** [comment]
yes @Cadair @hayesla  i guess we actually need to update example gallery  only thats causing all the test failures

**Comment by chouhanaryan** [comment]
Hi, is this issue still open? What else needs to be added to resolve this issue?

**Comment by nabobalis** [comment]
I am not sure. We did have an issue with the NOAA predict which is now fixed in 2.0.2.

I will close this until someone else chimes up.
